client/my-sites/plugins/plugins-list/test/index.jsx

```diff
@@ -2,65 +2,93 @@
+		{ sites: sitesObject, slug: 'jetpack', name: 'Jetpack' },
+	];
 
-			props = {
-				plugins,
-				header: 'Plugins',
-				selectedSite: sites[ 0 ],
-				selectedSiteId: sites[ 0 ].ID,
-				isPlaceholder: false,
-				pluginUpdateCount: plugins.length,
-				inProgressStatuses: [],
-				pluginsOnSites: {
-					hello: plugins[ 0 ],
-					jetpack: plugins[ 1 ],
-				},
-				allSites: sites,
-				hasManagePlugins: true,
-				siteIsAtomic: true,
-			};
-		} );
+	const props = {
+		plugins: pluginsList,
+		header: 'Plugins',
+		selectedSite: sites[ 0 ],
+		selectedSiteId: sites[ 0 ].ID,
+		isPlaceholder: false,
+		pluginUpdateCount: pluginsList.length,
+		inProgressStatuses: [],
+		pluginsOnSites: {
+			hello: pluginsList[ 0 ],
+			jetpack: pluginsList[ 1 ],
+		},
+		allSites: sites,
+		hasManagePlugins: true,
+		siteIsAtomic: true,
+		removePluginStatuses: () => {},
+	};
 
-		beforeEach( () => {
-			renderedPluginsList = shallow( <PluginsList { ...props } /> );
+	describe( 'rendering bulk actions', () => {
+		test( 'should select all plugins when toggled on (site is Atomic)', async () => {
+			const user = userEvent.setup();
+			render( <PluginsList { ...props } /> );
+			const toggleBtn = screen.queryByRole( 'button', { name: 'Edit All' } );
+			await user.click( toggleBtn );
+			// Jetpack is auto-managed on Atomic sites, therefore there's only 'Hello Dolly' in here
+			const boxes = screen.queryAllByRole( 'checkbox', { checked: true } );
+			expect( boxes ).toHaveLength( 2 );
+			expect( boxes[ 0 ] ).toHaveAttribute( 'title', 'Hello Dolly' );
 		} );
 
-		test( 'should be intialized with no selectedPlugins', () => {
-			expect( renderedPluginsList.state().selectedPlugins ).toEqual( {} );
-		} );
+		test( 'should select all plugins when toggled on (site is Jetpack)', async () => {
+			const user = userEvent.setup();
+			render( <PluginsList { ...props } siteIsAtomic={ false } siteIsJetpack /> );
+			const toggleBtn = screen.queryByRole( 'button', { name: 'Edit All' } );
+			await user.click( toggleBtn );
 
-		test( 'should select all plugins when toggled on', () => {
-			renderedPluginsList.instance().toggleBulkManagement();
-			expect( renderedPluginsList.state().selectedPlugins ).toEqual( {
-				hello: expect.anything(),
-				jetpack: expect.anything(),
+			pluginsList.forEach( ( plugin ) => {
+				const box = screen.queryByRole( 'checkbox', { name: plugin.name } );
+				expect( box ).toBeVisible();
+				expect( box ).toBeChecked();
 			} );
 		} );
 
-		test( 'should always reset to all selected when toggled on', () => {
-			renderedPluginsList.instance().togglePlugin( plugins[ 0 ] );
-			expect( Object.keys( renderedPluginsList.state().selectedPlugins ) ).toHaveLength( 1 );
+		test( 'should always reset to all selected when toggled on', async () => {
+			const getSelectedPluginBoxes = () =>
+				pluginsList
+					.map( ( plugin ) =>
+						screen.queryByRole( 'checkbox', { name: plugin.name, checked: true } )
```

Comment:
We could shorten this a bit:

```suggestion
					.map( ( { name } ) =>
						screen.queryByRole( 'checkbox', { name, checked: true } )
```